### ISSUE TITLE
Possible unclean shutdown of VM standalone and vmagent

### ISSUE THREAD

**Issue by aval13** [VictoriaMetrics/VictoriaMetrics#3685]
### Describe the bug

First of all, I am not entirely sure this is a bug in VictoriaMetrics Standalone or vmagent per se.
I have one server on which I run victoria-metrics-prod and vmagent-prod, and 2 other servers on which I have vmagent-prod.
All are Debian 10 (Buster), fully up to date, latest Debian 10 kernel running and no upgraded libraries not reloaded (after an update, I verify all the replaced libraries have been reloaded by depending services, so the servers are running the actual latest binaries).
This issue occurs only on one server which is running both victoria-metrics-prod and vmagent-prod.
When I do a restart of the victoriametrics or vmagent service, the logs produced on shutdown by the service end abruptly and systemd reports service was killed. On the other 2 servers, vmagent exists cleanly logging every step of the shutdown.
Even more confusing is that if I start a strace on the parent process (for both) and then issue a systemctl restart, both exit cleanly.
This is NOT a 1.86.2 issue, I noticed this since some versions ago (for vmagent), just never noticed until now that it also happens to VMS (unclean shutdown may break some data) or the "strace-voodoo".

### To Reproduce

I have no idea, only happens on one server.
All servers are bare metal servers, no VM, only the hardware is different between the one with issues and the other 2 with normal vmagents.
That's why I'm opening this ticket, maybe somebody has any idea how to investigate further if this is a Go issue, a VictoriaMetrics code issue, some Debian issue or hardware issue, maybe somebody else is affected by this or can pitch in some idea.
In the data below "zeon" is the affected server (VMS+vmagent), "pika" is a normal one (vmagent).

### Version

```
root@zeon:~# victoria-metrics-prod --version
victoria-metrics-20230118-200413-tags-v1.86.2-0-g9785b3f57
root@zeon:~# vmagent-prod --version
vmagent-20230118-200633-tags-v1.86.2-0-g9785b3f57
root@pika:~# vmagent-prod --version
vmagent-20230118-200633-tags-v1.86.2-0-g9785b3f57
```

### Logs

Affected vmagent shutdown logs (ISSUE):
```
Jan 21 16:45:00 zeon vmagent-prod[17549]: 2023-01-21T16:45:00.217Z        info        VictoriaMetrics/lib/promscrape/scraper.go:421        static_configs: added targets: 2, removed targets: 0; total targets: 2
Jan 21 16:45:53 zeon systemd[1]: Stopping VictoriaMetrics VMAgent...
Jan 21 16:45:53 zeon vmagent-prod[17549]: 2023-01-21T16:45:53.256Z        info        VictoriaMetrics/app/vmagent/main.go:131        received signal terminated
Jan 21 16:45:53 zeon systemd[1]: vmagent.service: Main process exited, code=killed, status=15/TERM
Jan 21 16:45:53 zeon systemd[1]: vmagent.service: Succeeded.
Jan 21 16:45:53 zeon systemd[1]: Stopped VictoriaMetrics VMAgent.
```

Another server's vmagent service shutdown (OK):
```
Jan 21 18:45:01 pika systemd[1]: Stopping VictoriaMetrics...
Jan 21 18:45:01 pika vmagent-prod[21152]: 2023-01-21T16:45:01.777Z        info        VictoriaMetrics/app/vmagent/main.go:131        received signal terminated
Jan 21 18:45:01 pika vmagent-prod[21152]: 2023-01-21T16:45:01.777Z        info        VictoriaMetrics/app/vmagent/main.go:135        gracefully shutting down webservice at "127.0.0.1:8429"
Jan 21 18:45:01 pika vmagent-prod[21152]: 2023-01-21T16:45:01.777Z        info        VictoriaMetrics/app/vmagent/main.go:139        successfully shut down the webservice in 0.000 seconds
Jan 21 18:45:01 pika vmagent-prod[21152]: 2023-01-21T16:45:01.778Z        info        VictoriaMetrics/lib/promscrape/config.go:211        stopping service discovery routines...
Jan 21 18:45:01 pika vmagent-prod[21152]: 2023-01-21T16:45:01.778Z        info        VictoriaMetrics/lib/promscrape/config.go:215        stopped service discovery routines in 0.000 seconds
Jan 21 18:45:01 pika vmagent-prod[21152]: 2023-01-21T16:45:01.778Z        info        VictoriaMetrics/lib/promscrape/scraper.go:186        stopping Prometheus scrapers
Jan 21 18:45:01 pika vmagent-prod[21152]: 2023-01-21T16:45:01.778Z        info        VictoriaMetrics/lib/promscrape/scraper.go:189        stopped Prometheus scrapers in 0.000 seconds
Jan 21 18:45:01 pika vmagent-prod[21152]: 2023-01-21T16:45:01.778Z        info        VictoriaMetrics/app/vmagent/remotewrite/client.go:175        stopped client for -remoteWrite.url="1:secret-url"
Jan 21 18:45:02 pika vmagent-prod[21152]: 2023-01-21T16:45:02.019Z        info        VictoriaMetrics/lib/persistentqueue/fastqueue.go:90        closed fast persistent queue at "/var/lib/vmagent-remotewrite-data/persistent-queue/1_0211BCEF6B85E01D"
Jan 21 18:45:02 pika vmagent-prod[21152]: 2023-01-21T16:45:02.019Z        info        VictoriaMetrics/app/vmagent/main.go:159        successfully stopped vmagent in 0.241 seconds
Jan 21 18:45:02 pika systemd[1]: vmagent.service: Succeeded.
Jan 21 18:45:02 pika systemd[1]: Stopped VictoriaMetrics.
```

Affected victoriametrics shutdown logs (ISSUE):
```
Jan 21 16:33:26 zeon victoria-metrics-prod[16724]: 2023-01-21T16:33:26.505Z        info        VictoriaMetrics/lib/httpserver/httpserver.go:97        pprof handlers are exposed at http://127.0.0.1:8428/debug/pprof/
Jan 21 16:59:26 zeon systemd[1]: Stopping VictoriaMetrics Standalone...
Jan 21 16:59:26 zeon victoria-metrics-prod[16724]: 2023-01-21T16:59:26.885Z        info        VictoriaMetrics/app/victoria-metrics/main.go:71        received signal terminated
Jan 21 16:59:26 zeon systemd[1]: victoriametrics.service: Main process exited, code=killed, status=15/TERM
Jan 21 16:59:26 zeon systemd[1]: victoriametrics.service: Succeeded.
Jan 21 16:59:26 zeon systemd[1]: Stopped VictoriaMetrics Standalone.
```

systemctl stop victoriametrics while running a strace on the main PID of victoria-metrics-prod process tree shutdown logs (STRANGELY OK):
```
Jan 21 17:00:15 zeon systemd[1]: Stopping VictoriaMetrics Standalone...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.374Z        info        VictoriaMetrics/app/victoria-metrics/main.go:71        received signal terminated
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.374Z        info        VictoriaMetrics/app/victoria-metrics/self_scraper.go:57        stopped self-scraping `/metrics` page
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.374Z        info        VictoriaMetrics/app/victoria-metrics/main.go:75        gracefully shutting down webservice at ":8428"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.375Z        info        VictoriaMetrics/app/victoria-metrics/main.go:81        successfully shut down the webservice in 0.000 seconds
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.375Z        info        VictoriaMetrics/app/vmstorage/main.go:244        gracefully closing the storage at /var/lib/victoriametrics
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.375Z        info        VictoriaMetrics/lib/storage/partition.go:774        waiting for service workers to stop on "/var/lib/victoriametrics/data/small/2023_01"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.375Z        info        VictoriaMetrics/lib/storage/partition.go:777        service workers stopped in 0.000 seconds on "/var/lib/victoriametrics/data/small/2023_01"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.375Z        info        VictoriaMetrics/lib/storage/partition.go:779        flushing inmemory parts to files on "/var/lib/victoriametrics/data/small/2023_01"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.575Z        info        VictoriaMetrics/lib/storage/partition.go:782        inmemory parts have been flushed to files in 0.200 seconds on "/var/lib/victoriametrics/data/small/2023_01"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.576Z        info        VictoriaMetrics/lib/storage/partition.go:774        waiting for service workers to stop on "/var/lib/victoriametrics/data/small/2022_07"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.576Z        info        VictoriaMetrics/lib/storage/partition.go:777        service workers stopped in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_07"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.576Z        info        VictoriaMetrics/lib/storage/partition.go:779        flushing inmemory parts to files on "/var/lib/victoriametrics/data/small/2022_07"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.576Z        info        VictoriaMetrics/lib/storage/partition.go:782        inmemory parts have been flushed to files in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_07"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.577Z        info        VictoriaMetrics/lib/storage/partition.go:774        waiting for service workers to stop on "/var/lib/victoriametrics/data/small/2022_11"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.577Z        info        VictoriaMetrics/lib/storage/partition.go:777        service workers stopped in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_11"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.577Z        info        VictoriaMetrics/lib/storage/partition.go:779        flushing inmemory parts to files on "/var/lib/victoriametrics/data/small/2022_11"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.577Z        info        VictoriaMetrics/lib/storage/partition.go:782        inmemory parts have been flushed to files in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_11"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.585Z        info        VictoriaMetrics/lib/storage/partition.go:774        waiting for service workers to stop on "/var/lib/victoriametrics/data/small/2022_12"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.585Z        info        VictoriaMetrics/lib/storage/partition.go:777        service workers stopped in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_12"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.585Z        info        VictoriaMetrics/lib/storage/partition.go:779        flushing inmemory parts to files on "/var/lib/victoriametrics/data/small/2022_12"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.586Z        info        VictoriaMetrics/lib/storage/partition.go:782        inmemory parts have been flushed to files in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_12"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.592Z        info        VictoriaMetrics/lib/storage/partition.go:774        waiting for service workers to stop on "/var/lib/victoriametrics/data/small/2022_08"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.592Z        info        VictoriaMetrics/lib/storage/partition.go:777        service workers stopped in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_08"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.592Z        info        VictoriaMetrics/lib/storage/partition.go:779        flushing inmemory parts to files on "/var/lib/victoriametrics/data/small/2022_08"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.592Z        info        VictoriaMetrics/lib/storage/partition.go:782        inmemory parts have been flushed to files in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_08"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.593Z        info        VictoriaMetrics/lib/storage/partition.go:774        waiting for service workers to stop on "/var/lib/victoriametrics/data/small/2022_09"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.593Z        info        VictoriaMetrics/lib/storage/partition.go:777        service workers stopped in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_09"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.593Z        info        VictoriaMetrics/lib/storage/partition.go:779        flushing inmemory parts to files on "/var/lib/victoriametrics/data/small/2022_09"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.593Z        info        VictoriaMetrics/lib/storage/partition.go:782        inmemory parts have been flushed to files in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_09"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.594Z        info        VictoriaMetrics/lib/storage/partition.go:774        waiting for service workers to stop on "/var/lib/victoriametrics/data/small/2022_10"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.594Z        info        VictoriaMetrics/lib/storage/partition.go:777        service workers stopped in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_10"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.594Z        info        VictoriaMetrics/lib/storage/partition.go:779        flushing inmemory parts to files on "/var/lib/victoriametrics/data/small/2022_10"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.594Z        info        VictoriaMetrics/lib/storage/partition.go:782        inmemory parts have been flushed to files in 0.000 seconds on "/var/lib/victoriametrics/data/small/2022_10"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.595Z        info        VictoriaMetrics/lib/mergeset/table.go:396        waiting for background workers to stop on "/var/lib/victoriametrics/indexdb/1731418C384B3FC6"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.595Z        info        VictoriaMetrics/lib/mergeset/table.go:399        background workers stopped in 0.000 seconds on "/var/lib/victoriametrics/indexdb/1731418C384B3FC6"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.595Z        info        VictoriaMetrics/lib/mergeset/table.go:401        flushing inmemory parts to files on "/var/lib/victoriametrics/indexdb/1731418C384B3FC6"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.952Z        info        VictoriaMetrics/lib/mergeset/table.go:404        inmemory parts have been successfully flushed to files in 0.357 seconds at "/var/lib/victoriametrics/indexdb/1731418C384B3FC6"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.952Z        info        VictoriaMetrics/lib/mergeset/table.go:406        waiting for flush callback worker to stop on "/var/lib/victoriametrics/indexdb/1731418C384B3FC6"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.952Z        info        VictoriaMetrics/lib/mergeset/table.go:409        flush callback worker stopped in 0.000 seconds on "/var/lib/victoriametrics/indexdb/1731418C384B3FC6"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.954Z        info        VictoriaMetrics/lib/mergeset/table.go:396        waiting for background workers to stop on "/var/lib/victoriametrics/indexdb/16EF4C01E87864B7"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.954Z        info        VictoriaMetrics/lib/mergeset/table.go:399        background workers stopped in 0.000 seconds on "/var/lib/victoriametrics/indexdb/16EF4C01E87864B7"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.954Z        info        VictoriaMetrics/lib/mergeset/table.go:401        flushing inmemory parts to files on "/var/lib/victoriametrics/indexdb/16EF4C01E87864B7"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.954Z        info        VictoriaMetrics/lib/mergeset/table.go:404        inmemory parts have been successfully flushed to files in 0.000 seconds at "/var/lib/victoriametrics/indexdb/16EF4C01E87864B7"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.954Z        info        VictoriaMetrics/lib/mergeset/table.go:406        waiting for flush callback worker to stop on "/var/lib/victoriametrics/indexdb/16EF4C01E87864B7"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.954Z        info        VictoriaMetrics/lib/mergeset/table.go:409        flush callback worker stopped in 0.000 seconds on "/var/lib/victoriametrics/indexdb/16EF4C01E87864B7"
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.957Z        info        VictoriaMetrics/lib/storage/storage.go:1025        saving MetricName->TSID cache to "/var/lib/victoriametrics/cache/metricName_tsid"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.967Z        info        VictoriaMetrics/lib/storage/storage.go:1032        saved MetricName->TSID cache to "/var/lib/victoriametrics/cache/metricName_tsid" in 0.010 seconds; entriesCount: 18041; sizeBy
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.968Z        info        VictoriaMetrics/lib/storage/storage.go:1025        saving MetricID->TSID cache to "/var/lib/victoriametrics/cache/metricID_tsid"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.969Z        info        VictoriaMetrics/lib/storage/storage.go:1032        saved MetricID->TSID cache to "/var/lib/victoriametrics/cache/metricID_tsid" in 0.001 seconds; entriesCount: 0; sizeBytes: 0
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.970Z        info        VictoriaMetrics/lib/storage/storage.go:1025        saving MetricID->MetricName cache to "/var/lib/victoriametrics/cache/metricID_metricName"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.972Z        info        VictoriaMetrics/lib/storage/storage.go:1032        saved MetricID->MetricName cache to "/var/lib/victoriametrics/cache/metricID_metricName" in 0.002 seconds; entriesCount: 0; si
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/lib/storage/storage.go:930        saving curr_hour_metric_ids to "/var/lib/victoriametrics/cache/curr_hour_metric_ids"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/lib/storage/storage.go:943        saved curr_hour_metric_ids to "/var/lib/victoriametrics/cache/curr_hour_metric_ids" in 0.000 seconds; entriesCount: 4717; sizeB
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/lib/storage/storage.go:930        saving prev_hour_metric_ids to "/var/lib/victoriametrics/cache/prev_hour_metric_ids"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/lib/storage/storage.go:943        saved prev_hour_metric_ids to "/var/lib/victoriametrics/cache/prev_hour_metric_ids" in 0.000 seconds; entriesCount: 7865; sizeB
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/lib/storage/storage.go:912        saving next_day_metric_ids to "/var/lib/victoriametrics/cache/next_day_metric_ids"...
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/lib/storage/storage.go:925        saved next_day_metric_ids to "/var/lib/victoriametrics/cache/next_day_metric_ids" in 0.000 seconds; entriesCount: 0; sizeBytes:
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/app/vmstorage/main.go:249        successfully closed the storage in 0.599 seconds
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/app/vmstorage/main.go:251        the storage has been stopped
Jan 21 17:00:15 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:15.973Z        info        VictoriaMetrics/app/vmselect/promql/rollup_result_cache.go:177        saving rollupResult cache to "/var/lib/victoriametrics/cache/rollupResult"...
Jan 21 17:00:16 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:16.019Z        info        VictoriaMetrics/app/vmselect/promql/rollup_result_cache.go:188        saved rollupResult cache to "/var/lib/victoriametrics/cache/rollupResult" in 0.046 seconds; entriesCount: 0
Jan 21 17:00:16 zeon victoria-metrics-prod[18436]: 2023-01-21T17:00:16.019Z        info        VictoriaMetrics/app/victoria-metrics/main.go:88        the VictoriaMetrics has been stopped in 0.645 seconds
Jan 21 17:00:16 zeon systemd[1]: victoriametrics.service: Succeeded.
Jan 21 17:00:16 zeon systemd[1]: Stopped VictoriaMetrics Standalone.

```

### Used command-line flags

```
root@zeon:~# ps -ef | grep victoria-metrics-prod
victori+ 18599     1  0 17:00 ?        00:00:09 /usr/local/sbin/victoria-metrics-prod -storageDataPath /var/lib/victoriametrics -retentionPeriod 6 -http.maxGracefulShutdownDuration 30s -selfScrapeInterval 60s

root@zeon:~# ps -ef | grep vmagent-prod
vmagent  17594     1  0 16:46 ?        00:00:01 /usr/local/sbin/vmagent-prod -httpListenAddr 127.0.0.1:8429 -promscrape.config /etc/vmagent/vmagent.conf -remoteWrite.maxDiskUsagePerURL 500000000 -remoteWrite.tmpDataPath /var/lib/vmagent-remotewrite-data -http.maxGracefulShutdownDuration 30s -remoteWrite.url http://localhost:8428/api/v1/write

root@pika:~# ps -ef | grep vmagent-prod
vmagent  21295     1  0 18:45 ?        00:00:04 /usr/local/sbin/vmagent-prod -httpListenAddr 127.0.0.1:8429 -promscrape.config /etc/vmagent/vmagent.conf -remoteWrite.maxDiskUsagePerURL 500000000 -remoteWrite.tmpDataPath /var/lib/vmagent-remotewrite-data -http.maxGracefulShutdownDuration 30s -remoteWrite.url http://10.1.0.22:8428/api/v1/write

```
### Additional information

Affected server CPU is an Intel Xeon W3520.
OK servers' CPUs are an Intel J1900 and an Intel i3-4170T.

```
root@zeon:~# uname -a
Linux zeon 4.19.0-23-amd64 #1 SMP Debian 4.19.269-1 (2022-12-20) x86_64 GNU/Linux
root@zeon:~# systemd --version
systemd 241 (241)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid
root@zeon:~# cat /etc/debian_version 
10.13

root@pika:~# uname -a
Linux pika 4.19.0-23-amd64 #1 SMP Debian 4.19.269-1 (2022-12-20) x86_64 GNU/Linux
root@pika:~# systemd --version
systemd 241 (241)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid
root@pika:~# cat /etc/debian_version 
10.13
```

All vmagents are controlled by a vmagent.service in /etc/systemd/system/, all unitfiles are the same (except for the VMS IP).
VMS is also started by a victoriametrics.service unitfile.
Will provide the unitfiles is needed.
The binaries are in /usr/local/sbin/ in the naming format vmagent-prod-VERSION and victoria-metrics-prod-VERSION, the unitfiles start /usr/local/sbin/vmagent-prod and /usr/local/sbin/victoria-metrics-prod which are symlinks to the actual version binary I want to start.

I also have a grafana-server running on that server (also written in Go as far as I remember), shuts down normally, no issues.
I cannot find any other service that is affected by this issue like VMS and vmagent.

**Comment by hagen1778** [comment]
I don't have much expertise here, but could it be the same issue as here https://github.com/Icinga/icinga2/issues/7291 ?

**Comment by valyala** [comment]
@zekker6 , could you try investigating this issue?

**Comment by zekker6** [comment]
@aval13 Could you please share `systemctl show` for both failing and normal units?
Also, a bit more log lines for failed vmagent would help. 

Need to see if `SIGHUP` have been sent to the process at all and if it was sent what was the delay between `SIGHUP` and `SIGKILL` afterwards.

Tried to reproduce it on several VMs and also directly on hw, but haven't found a way to reproduce this yet.

**Comment by aval13** [comment]
Hello @zekker6 , here are in order:
- systemctl show for vmagent on issue server (zeon) and ok server (pika)
- systemd unit file for vmagent (they are nearly identical except for the "where is victoria metrics" parameter)

show on issue server:
```
root@zeon:~# systemctl show vmagent | cat
Type=simple
Restart=on-failure
PIDFile=/run/vmagent/vmagent.pid
NotifyAccess=none
RestartUSec=10s
TimeoutStartUSec=1min 30s
TimeoutStopUSec=1min 30s
RuntimeMaxUSec=infinity
WatchdogUSec=0
WatchdogTimestampMonotonic=0
RootDirectoryStartOnly=no
RemainAfterExit=no
GuessMainPID=yes
MainPID=17594
ControlPID=0
FileDescriptorStoreMax=0
NFileDescriptorStore=0
StatusErrno=0
Result=success
UID=997
GID=996
NRestarts=0
ExecMainStartTimestamp=Sat 2023-01-21 16:46:18 UTC
ExecMainStartTimestampMonotonic=1893213139172
ExecMainExitTimestampMonotonic=0
ExecMainPID=17594
ExecMainCode=0
ExecMainStatus=0
ExecStart={ path=/usr/local/sbin/vmagent-prod ; argv[]=/usr/local/sbin/vmagent-prod -httpListenAddr 127.0.0.1:8429 -promscrape.config /etc/vmagent/vmagent.conf -remoteWrite.maxDiskUsagePerURL 500000000 -remoteWrite.tmpDataPath /var/lib/vmagent-remotewrite-data -http.maxGracefulShutdownDuration 30s -remoteWrite.url http://localhost:8428/api/v1/write ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
ExecStop={ path=/bin/kill ; argv[]=/bin/kill -s SIGTERM $MAINPID ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
Slice=system.slice
ControlGroup=/system.slice/vmagent.service
MemoryCurrent=36864000
CPUUsageNSec=[not set]
TasksCurrent=14
IPIngressBytes=18446744073709551615
IPIngressPackets=18446744073709551615
IPEgressBytes=18446744073709551615
IPEgressPackets=18446744073709551615
Delegate=no
CPUAccounting=no
CPUWeight=[not set]
StartupCPUWeight=[not set]
CPUShares=[not set]
StartupCPUShares=[not set]
CPUQuotaPerSecUSec=infinity
IOAccounting=no
IOWeight=[not set]
StartupIOWeight=[not set]
BlockIOAccounting=no
BlockIOWeight=[not set]
StartupBlockIOWeight=[not set]
MemoryAccounting=yes
MemoryMin=0
MemoryLow=0
MemoryHigh=infinity
MemoryMax=infinity
MemorySwapMax=infinity
MemoryLimit=infinity
DevicePolicy=auto
TasksAccounting=yes
TasksMax=4915
IPAccounting=no
UMask=0022
LimitCPU=infinity
LimitCPUSoft=infinity
LimitFSIZE=infinity
LimitFSIZESoft=infinity
LimitDATA=infinity
LimitDATASoft=infinity
LimitSTACK=infinity
LimitSTACKSoft=8388608
LimitCORE=infinity
LimitCORESoft=0
LimitRSS=infinity
LimitRSSSoft=infinity
LimitNOFILE=32000
LimitNOFILESoft=32000
LimitAS=infinity
LimitASSoft=infinity
LimitNPROC=32000
LimitNPROCSoft=32000
LimitMEMLOCK=65536
LimitMEMLOCKSoft=65536
LimitLOCKS=infinity
LimitLOCKSSoft=infinity
LimitSIGPENDING=63981
LimitSIGPENDINGSoft=63981
LimitMSGQUEUE=819200
LimitMSGQUEUESoft=819200
LimitNICE=0
LimitNICESoft=0
LimitRTPRIO=0
LimitRTPRIOSoft=0
LimitRTTIME=infinity
LimitRTTIMESoft=infinity
OOMScoreAdjust=0
Nice=0
IOSchedulingClass=0
IOSchedulingPriority=0
CPUSchedulingPolicy=0
CPUSchedulingPriority=0
TimerSlackNSec=50000
CPUSchedulingResetOnFork=no
NonBlocking=no
StandardInput=null
StandardInputData=
StandardOutput=journal
StandardError=inherit
TTYReset=no
TTYVHangup=no
TTYVTDisallocate=no
SyslogPriority=30
SyslogLevelPrefix=yes
SyslogLevel=6
SyslogFacility=3
LogLevelMax=-1
LogRateLimitIntervalUSec=0
LogRateLimitBurst=0
SecureBits=0
CapabilityBoundingSet=cap_chown cap_dac_override cap_dac_read_search cap_fowner cap_fsetid cap_kill cap_setgid cap_setuid cap_setpcap cap_linux_immutable cap_net_bind_service cap_net_broadcast cap_net_admin cap_net_raw cap_ipc_lock cap_ipc_owner cap_sys_module cap_sys_rawio cap_sys_chroot cap_sys_ptrace cap_sys_pacct cap_sys_admin cap_sys_boot cap_sys_nice cap_sys_resource cap_sys_time cap_sys_tty_config cap_mknod cap_lease cap_audit_write cap_audit_control cap_setfcap cap_mac_override cap_mac_admin cap_syslog cap_wake_alarm cap_block_suspend
AmbientCapabilities=
User=vmagent
Group=vmagent
DynamicUser=no
RemoveIPC=no
MountFlags=
PrivateTmp=no
PrivateDevices=no
ProtectKernelTunables=no
ProtectKernelModules=no
ProtectControlGroups=no
PrivateNetwork=no
PrivateUsers=no
PrivateMounts=no
ProtectHome=no
ProtectSystem=no
SameProcessGroup=no
UtmpMode=init
IgnoreSIGPIPE=yes
NoNewPrivileges=no
SystemCallErrorNumber=0
LockPersonality=no
RuntimeDirectoryPreserve=no
RuntimeDirectoryMode=0755
RuntimeDirectory=vmagent
StateDirectoryMode=0755
StateDirectory=vmagent-remotewrite-data
CacheDirectoryMode=0755
LogsDirectoryMode=0755
ConfigurationDirectoryMode=0755
MemoryDenyWriteExecute=no
RestrictRealtime=no
RestrictNamespaces=no
MountAPIVFS=no
KeyringMode=private
KillMode=control-group
KillSignal=15
FinalKillSignal=9
SendSIGKILL=yes
SendSIGHUP=no
WatchdogSignal=6
Id=vmagent.service
Names=vmagent.service
Requires=sysinit.target system.slice -.mount
WantedBy=multi-user.target
Conflicts=shutdown.target
Before=multi-user.target shutdown.target
After=system.slice sysinit.target systemd-journald.socket -.mount network.target basic.target
RequiresMountsFor=/run/vmagent /var/lib/vmagent-remotewrite-data
Description=VictoriaMetrics VMAgent
LoadState=loaded
ActiveState=active
SubState=running
FragmentPath=/etc/systemd/system/vmagent.service
UnitFileState=enabled
UnitFilePreset=enabled
StateChangeTimestamp=Sat 2023-01-21 16:46:18 UTC
StateChangeTimestampMonotonic=1893213139698
InactiveExitTimestamp=Sat 2023-01-21 16:46:18 UTC
InactiveExitTimestampMonotonic=1893213139698
ActiveEnterTimestamp=Sat 2023-01-21 16:46:18 UTC
ActiveEnterTimestampMonotonic=1893213139698
ActiveExitTimestamp=Sat 2023-01-21 16:46:18 UTC
ActiveExitTimestampMonotonic=1893213090848
InactiveEnterTimestamp=Sat 2023-01-21 16:46:18 UTC
InactiveEnterTimestampMonotonic=1893213136634
CanStart=yes
CanStop=yes
CanReload=no
CanIsolate=no
StopWhenUnneeded=no
RefuseManualStart=no
RefuseManualStop=no
AllowIsolate=no
DefaultDependencies=yes
OnFailureJobMode=replace
IgnoreOnIsolate=no
NeedDaemonReload=no
JobTimeoutUSec=infinity
JobRunningTimeoutUSec=infinity
JobTimeoutAction=none
ConditionResult=yes
AssertResult=yes
ConditionTimestamp=Sat 2023-01-21 16:46:18 UTC
ConditionTimestampMonotonic=1893213137955
AssertTimestamp=Sat 2023-01-21 16:46:18 UTC
AssertTimestampMonotonic=1893213137955
Transient=no
Perpetual=no
StartLimitIntervalUSec=1min
StartLimitBurst=3
StartLimitAction=none
FailureAction=none
FailureActionExitStatus=-1
SuccessAction=none
SuccessActionExitStatus=-1
InvocationID=cb14ee3a458842b2bf5001588abed4fb
CollectMode=inactive
```

show on ok server:
```
root@pika:~# systemctl show vmagent | cat
Type=simple
Restart=on-failure
PIDFile=/run/vmagent/vmagent.pid
NotifyAccess=none
RestartUSec=10s
TimeoutStartUSec=1min 30s
TimeoutStopUSec=1min 30s
RuntimeMaxUSec=infinity
WatchdogUSec=0
WatchdogTimestampMonotonic=0
RootDirectoryStartOnly=no
RemainAfterExit=no
GuessMainPID=yes
MainPID=21295
ControlPID=0
FileDescriptorStoreMax=0
NFileDescriptorStore=0
StatusErrno=0
Result=success
UID=103
GID=127
NRestarts=0
ExecMainStartTimestamp=Sat 2023-01-21 18:45:02 EET
ExecMainStartTimestampMonotonic=623121322458
ExecMainExitTimestampMonotonic=0
ExecMainPID=21295
ExecMainCode=0
ExecMainStatus=0
ExecStart={ path=/usr/local/sbin/vmagent-prod ; argv[]=/usr/local/sbin/vmagent-prod -httpListenAddr 127.0.0.1:8429 -promscrape.config /etc/vmagent/vmagent.conf -remoteWrite.maxDiskUsagePerURL 500000000 -remoteWrite.tmpDataPath /var/lib/vmagent-remotewrite-data -http.maxGracefulShutdownDuration 30s -remoteWrite.url http://10.1.0.22:8428/api/v1/write ; ignore_errors=no ; start_time=[Sat 2023-01-21 18:45:02 EET] ; stop_time=[n/a] ; pid=21295 ; code=(null) ; status=0/0 }
ExecStop={ path=/bin/kill ; argv[]=/bin/kill -s SIGTERM $MAINPID ; ignore_errors=no ; start_time=[n/a] ; stop_time=[n/a] ; pid=0 ; code=(null) ; status=0/0 }
Slice=system.slice
ControlGroup=/system.slice/vmagent.service
MemoryCurrent=40284160
CPUUsageNSec=[not set]
TasksCurrent=11
IPIngressBytes=18446744073709551615
IPIngressPackets=18446744073709551615
IPEgressBytes=18446744073709551615
IPEgressPackets=18446744073709551615
Delegate=no
CPUAccounting=no
CPUWeight=[not set]
StartupCPUWeight=[not set]
CPUShares=[not set]
StartupCPUShares=[not set]
CPUQuotaPerSecUSec=infinity
IOAccounting=no
IOWeight=[not set]
StartupIOWeight=[not set]
BlockIOAccounting=no
BlockIOWeight=[not set]
StartupBlockIOWeight=[not set]
MemoryAccounting=yes
MemoryMin=0
MemoryLow=0
MemoryHigh=infinity
MemoryMax=infinity
MemorySwapMax=infinity
MemoryLimit=infinity
DevicePolicy=auto
TasksAccounting=yes
TasksMax=4915
IPAccounting=no
UMask=0022
LimitCPU=infinity
LimitCPUSoft=infinity
LimitFSIZE=infinity
LimitFSIZESoft=infinity
LimitDATA=infinity
LimitDATASoft=infinity
LimitSTACK=infinity
LimitSTACKSoft=8388608
LimitCORE=infinity
LimitCORESoft=0
LimitRSS=infinity
LimitRSSSoft=infinity
LimitNOFILE=32000
LimitNOFILESoft=32000
LimitAS=infinity
LimitASSoft=infinity
LimitNPROC=32000
LimitNPROCSoft=32000
LimitMEMLOCK=65536
LimitMEMLOCKSoft=65536
LimitLOCKS=infinity
LimitLOCKSSoft=infinity
LimitSIGPENDING=62841
LimitSIGPENDINGSoft=62841
LimitMSGQUEUE=819200
LimitMSGQUEUESoft=819200
LimitNICE=0
LimitNICESoft=0
LimitRTPRIO=0
LimitRTPRIOSoft=0
LimitRTTIME=infinity
LimitRTTIMESoft=infinity
OOMScoreAdjust=0
Nice=0
IOSchedulingClass=0
IOSchedulingPriority=0
CPUSchedulingPolicy=0
CPUSchedulingPriority=0
TimerSlackNSec=50000
CPUSchedulingResetOnFork=no
NonBlocking=no
StandardInput=null
StandardInputData=
StandardOutput=journal
StandardError=inherit
TTYReset=no
TTYVHangup=no
TTYVTDisallocate=no
SyslogPriority=30
SyslogLevelPrefix=yes
SyslogLevel=6
SyslogFacility=3
LogLevelMax=-1
LogRateLimitIntervalUSec=0
LogRateLimitBurst=0
SecureBits=0
CapabilityBoundingSet=cap_chown cap_dac_override cap_dac_read_search cap_fowner cap_fsetid cap_kill cap_setgid cap_setuid cap_setpcap cap_linux_immutable cap_net_bind_service cap_net_broadcast cap_net_admin cap_net_raw cap_ipc_lock cap_ipc_owner cap_sys_module cap_sys_rawio cap_sys_chroot cap_sys_ptrace cap_sys_pacct cap_sys_admin cap_sys_boot cap_sys_nice cap_sys_resource cap_sys_time cap_sys_tty_config cap_mknod cap_lease cap_audit_write cap_audit_control cap_setfcap cap_mac_override cap_mac_admin cap_syslog cap_wake_alarm cap_block_suspend
AmbientCapabilities=
User=vmagent
Group=vmagent
DynamicUser=no
RemoveIPC=no
MountFlags=
PrivateTmp=no
PrivateDevices=no
ProtectKernelTunables=no
ProtectKernelModules=no
ProtectControlGroups=no
PrivateNetwork=no
PrivateUsers=no
PrivateMounts=no
ProtectHome=no
ProtectSystem=no
SameProcessGroup=no
UtmpMode=init
IgnoreSIGPIPE=yes
NoNewPrivileges=no
SystemCallErrorNumber=0
LockPersonality=no
RuntimeDirectoryPreserve=no
RuntimeDirectoryMode=0755
RuntimeDirectory=vmagent
StateDirectoryMode=0755
StateDirectory=vmagent-remotewrite-data
CacheDirectoryMode=0755
LogsDirectoryMode=0755
ConfigurationDirectoryMode=0755
MemoryDenyWriteExecute=no
RestrictRealtime=no
RestrictNamespaces=no
MountAPIVFS=no
KeyringMode=private
KillMode=control-group
KillSignal=15
FinalKillSignal=9
SendSIGKILL=yes
SendSIGHUP=no
WatchdogSignal=6
Id=vmagent.service
Names=vmagent.service
Requires=sysinit.target -.mount system.slice
WantedBy=multi-user.target
Conflicts=shutdown.target
Before=multi-user.target shutdown.target
After=sysinit.target system.slice basic.target network.target -.mount systemd-journald.socket
RequiresMountsFor=/run/vmagent /var/lib/vmagent-remotewrite-data
Description=VictoriaMetrics
LoadState=loaded
ActiveState=active
SubState=running
FragmentPath=/etc/systemd/system/vmagent.service
UnitFileState=enabled
UnitFilePreset=enabled
StateChangeTimestamp=Sat 2023-01-21 18:45:02 EET
StateChangeTimestampMonotonic=623121323473
InactiveExitTimestamp=Sat 2023-01-21 18:45:02 EET
InactiveExitTimestampMonotonic=623121323473
ActiveEnterTimestamp=Sat 2023-01-21 18:45:02 EET
ActiveEnterTimestampMonotonic=623121323473
ActiveExitTimestamp=Sat 2023-01-21 18:45:01 EET
ActiveExitTimestampMonotonic=623121058352
InactiveEnterTimestamp=Sat 2023-01-21 18:45:02 EET
InactiveEnterTimestampMonotonic=623121316611
CanStart=yes
CanStop=yes
CanReload=no
CanIsolate=no
StopWhenUnneeded=no
RefuseManualStart=no
RefuseManualStop=no
AllowIsolate=no
DefaultDependencies=yes
OnFailureJobMode=replace
IgnoreOnIsolate=no
NeedDaemonReload=no
JobTimeoutUSec=infinity
JobRunningTimeoutUSec=infinity
JobTimeoutAction=none
ConditionResult=yes
AssertResult=yes
ConditionTimestamp=Sat 2023-01-21 18:45:02 EET
ConditionTimestampMonotonic=623121319580
AssertTimestamp=Sat 2023-01-21 18:45:02 EET
AssertTimestampMonotonic=623121319582
Transient=no
Perpetual=no
StartLimitIntervalUSec=1min
StartLimitBurst=3
StartLimitAction=none
FailureAction=none
FailureActionExitStatus=-1
SuccessAction=none
SuccessActionExitStatus=-1
InvocationID=a28f26702a24476abc40533e984d327d
CollectMode=inactive
```

systemd unit file
```
[Unit]
Description=VictoriaMetrics VMAgent
After=network.target
StartLimitBurst=3
StartLimitInterval=60

[Service]
Type=simple
User=vmagent
Group=vmagent
Restart=on-failure
RestartSec=10
RuntimeDirectory=vmagent
StateDirectory=vmagent-remotewrite-data
PIDFile=/run/vmagent/vmagent.pid
ExecStart=/usr/local/sbin/vmagent-prod -httpListenAddr "127.0.0.1:8429" -promscrape.config /etc/vmagent/vmagent.conf -remoteWrite.maxDiskUsagePerURL 500000000 -remoteWrite.tmpDataPath /var/lib/vmagent-remotewrite-data -http.maxGracefulShutdownDuration 30s -remoteWrite.url http://localhost:8428/api/v1/write
ExecStop=/bin/kill -s SIGTERM $MAINPID
LimitNOFILE=32000
LimitNPROC=32000

[Install]
WantedBy=multi-user.target
```

Just did a restart on the issue server for vmagent and everything shut down perfectly.
Then I did another restart and issue happened again.
Here are the logs from a few minutes ago for journalctl -b -u vmagent:
```
-- Logs begin at Thu 2023-02-16 20:43:49 UTC, end at Tue 2023-02-21 20:51:12 UTC. --
Feb 21 20:49:31 zeon systemd[1]: Stopping VictoriaMetrics VMAgent...
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.486Z        info        VictoriaMetrics/app/vmagent/main.go:131        received signal terminated
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.486Z        info        VictoriaMetrics/app/vmagent/main.go:135        gracefully shutting down webservice at "127.0.0.1:8429"
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.486Z        info        VictoriaMetrics/app/vmagent/main.go:139        successfully shut down the webservice in 0.000 seconds
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.486Z        info        VictoriaMetrics/lib/promscrape/config.go:211        stopping service discovery routines...
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.486Z        info        VictoriaMetrics/lib/promscrape/config.go:215        stopped service discovery routines in 0.000 seconds
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.486Z        info        VictoriaMetrics/lib/promscrape/scraper.go:186        stopping Prometheus scrapers
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.486Z        info        VictoriaMetrics/lib/promscrape/scraper.go:189        stopped Prometheus scrapers in 0.000 seconds
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.487Z        info        VictoriaMetrics/app/vmagent/remotewrite/client.go:175        stopped client for -remoteWrite.url="1:secret
-url"
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.535Z        info        VictoriaMetrics/lib/persistentqueue/fastqueue.go:90        closed fast persistent queue at "/var/lib/vmage
nt-remotewrite-data/persistent-queue/1_FF9C8EEDCB08798A"
Feb 21 20:49:31 zeon vmagent-prod[17594]: 2023-02-21T20:49:31.535Z        info        VictoriaMetrics/app/vmagent/main.go:159        successfully stopped vmagent in 0.049 seconds
Feb 21 20:49:31 zeon systemd[1]: vmagent.service: Succeeded.
Feb 21 20:49:31 zeon systemd[1]: Stopped VictoriaMetrics VMAgent.
Feb 21 20:49:31 zeon systemd[1]: Started VictoriaMetrics VMAgent.
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/lib/logger/flag.go:12        build version: vmagent-20230118-200633-tags-v1.86.2-0-g9785b3f57
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/lib/logger/flag.go:13        command-line flags
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/lib/logger/flag.go:20          -http.maxGracefulShutdownDuration="30s"
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/lib/logger/flag.go:20          -httpListenAddr="127.0.0.1:8429"
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/lib/logger/flag.go:20          -promscrape.config="/etc/vmagent/vmagent.conf"
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/lib/logger/flag.go:20          -remoteWrite.maxDiskUsagePerURL="500000000"
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/lib/logger/flag.go:20          -remoteWrite.tmpDataPath="/var/lib/vmagent-remotewrite-data"
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/lib/logger/flag.go:20          -remoteWrite.url="secret"
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.551Z        info        VictoriaMetrics/app/vmagent/main.go:102        starting vmagent at "127.0.0.1:8429"...
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.552Z        info        VictoriaMetrics/lib/memory/memory.go:42        limiting caches to 10086162432 bytes, leaving 6724108288 bytes to the OS according to -memory.allowedPercent=60
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.564Z        info        VictoriaMetrics/lib/persistentqueue/fastqueue.go:59        opened fast persistent queue at "/var/lib/vmagent-remotewrite-data/persistent-queue/1_FF9C8EEDCB08798A" with maxInmemoryBlocks=1600, it contains 0 pending bytes
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.564Z        info        VictoriaMetrics/app/vmagent/remotewrite/client.go:169        initialized client for -remoteWrite.url="1:secret-url"
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.564Z        info        VictoriaMetrics/app/vmagent/main.go:128        started vmagent in 0.013 seconds
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.564Z        info        VictoriaMetrics/lib/promscrape/scraper.go:109        reading Prometheus configs from "/etc/vmagent/vmagent.conf"
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.564Z        info        VictoriaMetrics/lib/httpserver/httpserver.go:96        starting http server at http://127.0.0.1:8429/
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.564Z        info        VictoriaMetrics/lib/httpserver/httpserver.go:97        pprof handlers are exposed at http://127.0.0.1:8429/debug/pprof/
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.566Z        info        VictoriaMetrics/lib/promscrape/config.go:120        starting service discovery routines...
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.566Z        info        VictoriaMetrics/lib/promscrape/config.go:126        started service discovery routines in 0.000 seconds
Feb 21 20:49:31 zeon vmagent-prod[22529]: 2023-02-21T20:49:31.567Z        info        VictoriaMetrics/lib/promscrape/scraper.go:421        static_configs: added targets: 2, removed targets: 0; total targets: 2
Feb 21 20:51:12 zeon systemd[1]: Stopping VictoriaMetrics VMAgent...
Feb 21 20:51:12 zeon vmagent-prod[22529]: 2023-02-21T20:51:12.102Z        info        VictoriaMetrics/app/vmagent/main.go:131        received signal terminated
Feb 21 20:51:12 zeon vmagent-prod[22529]: 2023-02-21T20:51:12.102Z        info        VictoriaMetrics/app/vmagent/main.go:135        gracefully shutting down webservice at "127.0.0.1:8429"
Feb 21 20:51:12 zeon systemd[1]: vmagent.service: Main process exited, code=killed, status=15/TERM
Feb 21 20:51:12 zeon systemd[1]: vmagent.service: Succeeded.
Feb 21 20:51:12 zeon systemd[1]: Stopped VictoriaMetrics VMAgent.
Feb 21 20:51:12 zeon systemd[1]: Started VictoriaMetrics VMAgent.
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/lib/logger/flag.go:12        build version: vmagent-20230118-200633-tags-v1.86.2-0-g9785b3f57
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/lib/logger/flag.go:13        command-line flags
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/lib/logger/flag.go:20          -http.maxGracefulShutdownDuration="30s"
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/lib/logger/flag.go:20          -httpListenAddr="127.0.0.1:8429"
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/lib/logger/flag.go:20          -promscrape.config="/etc/vmagent/vmagent.conf"
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/lib/logger/flag.go:20          -remoteWrite.maxDiskUsagePerURL="500000000"
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/lib/logger/flag.go:20          -remoteWrite.tmpDataPath="/var/lib/vmagent-remotewrite-data"
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/lib/logger/flag.go:20          -remoteWrite.url="secret"
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.116Z        info        VictoriaMetrics/app/vmagent/main.go:102        starting vmagent at "127.0.0.1:8429"...
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.117Z        info        VictoriaMetrics/lib/memory/memory.go:42        limiting caches to 10086162432 bytes, leaving 6724108288 bytes to the OS according to -memory.allowedPercent=60
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.118Z        info        VictoriaMetrics/lib/persistentqueue/fastqueue.go:59        opened fast persistent queue at "/var/lib/vmagent-remotewrite-data/persistent-queue/1_FF9C8EEDCB08798A" with maxInmemoryBlocks=1600, it contains 0 pending bytes
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.118Z        info        VictoriaMetrics/app/vmagent/remotewrite/client.go:169        initialized client for -remoteWrite.url="1:secret-url"
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.119Z        info        VictoriaMetrics/app/vmagent/main.go:128        started vmagent in 0.002 seconds
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.119Z        info        VictoriaMetrics/lib/promscrape/scraper.go:109        reading Prometheus configs from "/etc/vmagent/vmagent.conf"
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.119Z        info        VictoriaMetrics/lib/httpserver/httpserver.go:96        starting http server at http://127.0.0.1:8429/
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.119Z        info        VictoriaMetrics/lib/httpserver/httpserver.go:97        pprof handlers are exposed at http://127.0.0.1:8429/debug/pprof/
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.122Z        info        VictoriaMetrics/lib/promscrape/config.go:120        starting service discovery routines...
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.122Z        info        VictoriaMetrics/lib/promscrape/config.go:126        started service discovery routines in 0.000 seconds
Feb 21 20:51:12 zeon vmagent-prod[22647]: 2023-02-21T20:51:12.124Z        info        VictoriaMetrics/lib/promscrape/scraper.go:421        static_configs: added targets: 2, removed targets: 0; total targets: 2
```

I have also reverified and can confirm (after trying 3 times in a row) that running a strace on the vmagent-prod process has NEVER produced a unclean shutdown, it ALWAYS terminates gracefully. I have also observed this on victoriametrics standalone (tried only once, I feel confident can be reproduced reliably on this issue server).
```
strace -f -p `pidof vmagent-prod`
```

I am also very annoyed that this happens ONLY on one physical server, not on the other 2 (i do not have a vmagent on a VM).
If you have any idea on how to further debug or WHY the strace "guarantees" a clean shutdown, please let me know.
(I love investigating voodoos, but this is a bit out of my league).
The only hunch I still have but no idea how to verify is that somehow this is related to the CPU of the issue server.
```
model name      : Intel(R) Xeon(R) CPU           W3520  @ 2.67GHz
```

Thank you.

**Comment by aval13** [comment]
Hello @zekker6 , any ideas on what else I could test?

Just a headsup, soon-ish (1-2-3-something weeks) I will upgrade the servers to Debian 11 (bullseye) and also upgrade the VM versions.
So until then I can still test with the currently installed versions on Debian 10.

After the upgrades (whenever they will happen), I will update here if the issue still persists.
If it does not, maybe it was something related to Debian 10.
If it does, then the only idea I have it is that there is something in the hardware. Even if so, I would still be available to test anything we may come up with just to make sure we covered all the possibilities.

Thank you.

**Comment by zekker6** [comment]
@aval13 Sorry for the delay, missed notification from this one somehow.

The only way for this to happen is if vm process would not receive `SIGHUP` and won't do graceful shutdown untill `SIGKILL` will shut it down. As systemd units look very similar I could only suggest that the difference should be in systemd/cgroups versions and a way they are passing signals to the processes.

Did you already ran an upgrade and/or is issue still persisting there?

**Comment by hagen1778** [comment]
Closing the issue due to lack of activity. @aval13 feel free to re-open if the problem still exists
